### DISCUSSION TITLE
Feature Request - Synced Blocks

### DISCUSSION THREAD

**Discussion by roovyshapiro** [outline/outline#3340] [Ideas]
Notion has this very useful feature called [Synced Blocks](https://www.notion.so/help/synced-blocks)
It allows you to have a section of content which can be copy and pasted to multiple pages. When one of these "blocks" are edited, all of these changes are propagated to all synced blocks in near real-time.

Here's a GIF showing how this works in practice.

https://i.imgur.com/B7jVFT8.gif
![](https://i.imgur.com/B7jVFT8.gif)

This would be very helpful to have in Outline.
I'm writing sections of documents that I'm having trouble figuring out how to organize properly since they are referencing content in multiple different pages. I don't really want to copy and paste the same content into multiple pages since I might edit one and forget to edit the second one later.
What I do now as a workaround is just go into the other page and link the page and write "see here for more information" or something like that.

It's ok, but it would be nice to instead have a synced block which can be accessible on multiple pages.

_(this is a neat feature, but probably not a high priority task)_

**Comment by almereyda** [comment]
In other places, this is called https://en.wikipedia.org/wiki/Transclusion

LogSeq has a similar method to reference or embed arbitrary blocks¹.

---

¹ While they do not follow the blockprotocol.org, it appears useful to mention it here, too, as a generalised abstraction for (editable) block content.

**Comment by michaelkarrer81** [comment]
Great feature! DRY anyone ;)

**Comment by ThinkSalat** [comment]
This would be such a killer feature for my own system of PKM. Would be nice to have the ability to embed entire pages so one can access anything without navigation, unless they want to be entirely focused on that one thing. Considering lots of PKMs try to address the use case of having just a quick thought and wanting to throw it into your notes as quickly as possible, this would make that so much easier.

**Comment by zeekhuge** [comment]
There is [this PR to add `IFrame`s](https://github.com/outline/outline/pull/5137). By using `IFrame` for an outline document itself will probably solve this problem.

Although, not sure how permissions will be managed in such a case.

**Reply by ThinkSalat** [reply]
> There is [this PR to add `IFrame`s](https://github.com/outline/outline/pull/5137). By using `IFrame` for an outline document itself will probably solve this problem.
> 
> Although, not sure how permissions will be managed in such a case.

Good looking out! I'll be following this as it seems to be a fairly fluid solution.

**Comment by almereyda** [comment]
While Outline (currently?) cannot perform to be and does not aspire to be low code database, like Grist et al.: AppFlowy and AnyType show good progress on contesting Notion and Tana, many projects went into less open directions, like Affine or Budibase, and Outline is sufficiently webby to serve user bases well. AppFlowy is also appealing for its Y CRDT and conventionally accessible SQL store, while AnyType shows a lot more flexibility, yet builds on MongoDB. And both are not on the web.

<details><summary>Reference issues about a no code graph with spreadsheet interface</summary>
<p>

- https://github.com/outline/outline/discussions/3440
- https://github.com/outline/outline/discussions/6910
- https://github.com/outline/outline/discussions/6507

</p>
</details> 

I'd encourage self-hosters and collaboration application developers to give AppFlowy another try, since its Cloud implementation is now sufficiently matured to be used.

- [[FR] Documented production example behind a reverse proxy, here Træfik · Issue #622 · AppFlowy-IO/AppFlowy-Cloud](https://github.com/AppFlowy-IO/AppFlowy-Cloud/issues/622)

Since #3000 changed the data model of documents in a fundamental way, the use case described in the original post here becomes closer within reach again.

We already see partial display of documents, when hovering an internal link #6774:

![grafik](https://github.com/outline/outline/assets/1645308/35175da7-61fa-42ee-a663-1c356f493679)

Eventually this existing facility could be extended and repurposed to also allow selective display and writing to parts of documents?
